recipes/grpc/all/conanfile.py

```diff
@@ -0,0 +1,232 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class grpcConan(ConanFile):
+    name = "grpc"
+    description = "Google's RPC (remote procedure call) library and framework."
+    topics = ("conan", "grpc", "rpc")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/grpc/grpc"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    short_paths = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    # TODO: Add shared option
+    options = {
+        "fPIC": [True, False],
+        "build_codegen": [True, False],
+        "build_csharp_ext": [True, False],
+        "build_cpp_plugin": [True, False],
+        "build_csharp_plugin": [True, False],
+        "build_node_plugin": [True, False],
+        "build_objective_c_plugin": [True, False],
+        "build_php_plugin": [True, False],
+        "build_python_plugin": [True, False],
+        "build_ruby_plugin": [True, False]
```

Comment:
When using this recipe, the `build_` prefix does not make sense.
So I would replace it with `enable_`.
Also, what is the difference between `build_csharp_ext` and `build_csharp_plugin`?

Reply:
The extension is a shared library to be loaded by C# runtime, the plugin is an executable like the other plugins.

---

I agree `build_` is not the best from the consumer perspective (binaries are already built), but not sure if it is better to use `enable_` or just nothing:

```
[requires]
grpc/1.37.0

[options]
grpc:cpp_plugin=True
```

vs

```
[requires]
grpc/1.37.0

[options]
grpc:enable_cpp_plugin=True
```

Reply:
I like without enable (and without build).